lbrynet/lbryfilemanager/EncryptedFileManager.py

```diff
@@ -97,64 +113,75 @@ def _iter_streams(stream_hashes):
         yield defer.DeferredList(list(_iter_streams(stream_hashes)))
 
     @defer.inlineCallbacks
-    def _start_lbry_files(self):
-        def set_options_and_restore(rowid, stream_hash, options):
-            b_prm = self.session.base_payment_rate_manager
-            payment_rate_manager = NegotiatedPaymentRateManager(b_prm, self.session.blob_tracker)
-
-            d = self.start_lbry_file(rowid, stream_hash, payment_rate_manager,
-                                     blob_data_rate=options)
-            d.addCallback(lambda downloader: downloader.restore())
-            return d
-
-        def log_error(err, rowid, stream_hash, options):
-            log.error("An error occurred while starting a lbry file: %s", err.getErrorMessage())
-            log.debug(rowid)
-            log.debug(stream_hash)
-            log.debug(options)
-
-        def start_lbry_files(lbry_files_and_options):
-            for rowid, stream_hash, options in lbry_files_and_options:
-                d = set_options_and_restore(rowid, stream_hash, options)
-                d.addErrback(lambda err: log_error(err, rowid, stream_hash, options))
-            log.info("Started %i lbry files", len(self.lbry_files))
-            return True
+    def _restore_lbry_file(self, lbry_file):
+        try:
+            yield lbry_file.restore()
+        except Exception as err:
+            log.error("Failed to start stream: %s, error: %s", lbry_file.stream_hash, err)
+            self.lbry_files.remove(lbry_file)
+            # TODO: delete stream without claim instead of just removing from manager?
 
+    @defer.inlineCallbacks
+    def _start_lbry_files(self):
+        b_prm = self.session.base_payment_rate_manager
+        payment_rate_manager = NegotiatedPaymentRateManager(b_prm, self.session.blob_tracker)
         yield self._check_stream_info_manager()
-        files_and_options = yield self._get_all_lbry_files()
-        yield start_lbry_files(files_and_options)
+        lbry_files_and_options = yield self._get_all_lbry_files()
+        for rowid, stream_hash, options in lbry_files_and_options:
+            lbry_file = yield self.start_lbry_file(rowid, stream_hash, payment_rate_manager,
+                                                   blob_data_rate=options)
+            d = self._restore_lbry_file(lbry_file)
```

Comment:
mentioned in our phone chat - but this would be nice to have inside a deferredList so that we can keep track of the deferred instead of having it disappear into twisted.